api/ttn/lorawan/v3/qrcodegenerator.proto

```diff
@@ -130,3 +131,53 @@ service EndDeviceQRCodeGenerator {
     };
   }
 }
+
+message ParseGatewayQRCodeRequest {
+  // QR code format identifier.
+  // If this field is not specified, the server will default to ttigpro1.
+  string format_id = 1 [(validate.rules).string = {
+    pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$|^$",
+    max_len: 36
+  }];
+  // Raw QR code contents.
+  bytes qr_code = 2 [(validate.rules).bytes = {
+    min_len: 10,
+    max_len: 1024
+  }];
+}
+
+message ParseGatewayQRCodeResponse {
+  // Identifier of the format used to successfully parse the QR code data.
+  string format_id = 1;
```

Comment:
This needs at least a length validation, most likely an ID validation.

Reply:
This is also a response. Should it have validation on it? The request message where the format_id is specified does have a validation already:
```proto
message ParseGatewayQRCodeRequest {
  // QR code format identifier.
  // If this field is not specified, the server will default to ttigpro1.
  string format_id = 1 [(validate.rules).string = {
    pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$|^$",
    max_len: 36
  }];
  // Raw QR code contents.
  bytes qr_code = 2 [(validate.rules).bytes = {
    min_len: 10,
    max_len: 1024
  }];
}
```